packages/flutter_tools/lib/src/drive/drive_service.dart

```diff
@@ -242,14 +256,30 @@ class FlutterDriverService extends DriverService {
     bool androidEmulator,
     int driverPort,
     List<String> browserDimension,
+    String profileMemory,
   }) async {
-    return _processUtils.stream(<String>[
-      _dartSdkPath,
-      ...<String>[...arguments, testFile, '-rexpanded'],
-    ], environment: <String, String>{
-      'VM_SERVICE_URL': _vmServiceUri,
-      ...environment,
-    });
+    if (profileMemory != null) {
+      unawaited(_devtoolsLauncher.launch(
+        Uri.parse(_vmServiceUri),
+        additionalArguments: <String>['--profile-memory=$profileMemory'],
+      ));
+      // When profiling memory the original launch future will never complete.
```

Comment:
Why is this? Does the process get leaked?

Reply:
The process doesn't leak (see below where we make sure to kill it), but it doesn't behave the same with the "devtools serving on X", instead it should start writing out a memory profile